File: backend/app/services/recommendation_service.py
from app.schemas.scheme import Profile
from app.services.scheme_service import SCHEMES
from app.services.eligibility_service import evaluate
def recommend(profile: Profile):
    results=[]
    for scheme in SCHEMES:
        result=evaluate(scheme, profile)
        relevance=result['score']
        if profile.occupation and profile.occupation in scheme.eligibility.get('occupations',[]): relevance=max(relevance, 80)
        results.append({"scheme":scheme, "match_score":relevance, "why":[f"Your {m} matches the available scheme rule." for m in result['matched_conditions']] or ["This scheme may be relevant to your situation; check the official criteria."], "eligibility":result})
    return sorted(results,key=lambda x:x['match_score'],reverse=True)

File: lib/services/recommendation_service.ts
import { Profile, RecommendationResult } from '../types';
import { ALL_SCHEMES } from './scheme_service';
import { evaluateEligibility } from './eligibility_service';

export function recommendSchemes(profile: Profile): RecommendationResult[] {
  const results: RecommendationResult[] = [];
  for (const scheme of ALL_SCHEMES) {
    if (!scheme.is_active) continue;
    const result = evaluateEligibility(scheme, profile);
    let relevance = result.score;
    const occupations = scheme.eligibility?.occupations;
    if (profile.occupation && Array.isArray(occupations) && occupations.includes(profile.occupation)) {
      relevance = Math.max(relevance, 80);
    }

    const why =
      result.matched_conditions.length > 0
        ? result.matched_conditions.map(m => `Your ${m} matches the available scheme rule.`)
        : ['This scheme may be relevant to your situation; check the official criteria.'];

    results.push({
      scheme,
      match_score: relevance,
      why,
      eligibility: result,
    });
  }

  return results.sort((a, b) => b.match_score - a.match_score);
}
